### PR TITLE
[App Service] Fixes # 9887 webapp and functionapp, support assigning/removing user managed identity

### PR DESCRIPTION
**Pull request by cxznmhdcxz**
**Description**  

why this pr: https://github.com/Azure/azure-cli/issues/9887, https://github.com/Azure/azure-cli/issues/9887#issuecomment-597195457

Before this update:
`az webapp/functionapp identity assign` can only enable system identity on a webapp/functionapp.
`az webapp/functionapp identity remove` can only remove all identities on a webapp/functionapp.
After this update:
`az webapp/functionapp identity assign` supports adding user managed identity using `--identities` parameters.
`az webapp/functionapp identity remove` removes specific identities using `--identities` parameters

**Testing Guide**  

##### assign
- `az webapp identity assign -n app -g group` or
`az webapp identity assign -n app -g group --identities [system]` 
This should enable system identity on a webapp/functionapp, without affecting all the other identities are already assigned.

- `az webapp identity assign -n app -g group --identities identity` or
`az webapp identity assign -n app -g group --identities identity1 identity2 ...`
 This should assign one/multiple user managed identities to the webapp/functionapp, without affecting all the other identities 
 that are already assigned. 

##### remove
- `az webapp identity remove -n app -g group` or
`az webapp identity remove -n app -g group --identities [system]`
This should disable system identity on an app.

- `az webapp identity assign -n app -g group --identities [system] identity1`
This should disable system identity and a user managed identity.

The identities parameters are full ARM ids of user managed identities, or `[system]` to stand for system managed identity.
Assigning identities that are already assigned will do nothing.
If an app currently has no identities, `identity remove` will do nothing no matter what parameters.
Removing a user identity which is not assigned will raise error.
It's ok to remove system identity even if it's already disabled, it'll just do nothing.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
webapp

**Comment by yungezz** [comment]
hi @qwordy could you pls review this PR? thanks

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -637,14 +637,17 @@
 
 helps['functionapp identity assign'] = """
 type: command
-short-summary: assign or disable managed service identity to the web app
+short-summary: assign managed service identity to the web app
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:640]: user cannot disable MSI ?

Comment (cxznmhdcxz) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py:640]: `az functionapp/webapp identity remove` disables all managed identities on an app.

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py:640]: got it

Comment (cxznmhdcxz) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py:640]: Before this update, `identity assign` makes an app:
whatever identities are assigend -> only system managed identity assigned

**Comment by tzhanl** [comment]
@jongio for notification.

**Comment by panchagnula** [comment]
@ankitkumarr for renewing functionapp related changes

**Comment by panchagnula** [comment]
@btardif FYI

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 2]
```diff
@@ -1581,23 +1586,28 @@
 
 helps['webapp identity assign'] = """
 type: command
-short-summary: assign or disable managed service identity to the web app
+short-summary: assign managed service identity to the web app
 examples:
   - name: assign local identity and assign a reader role to the current resource group.
     text: >
         az webapp identity assign -g MyResourceGroup -n MyUniqueApp --role reader --scope /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/MyResourceGroup
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1593]: Is this example still accurate? is the identity defaulted to system-assigned?

Comment (cxznmhdcxz) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py:1593]: Yes, default is system-assigned.

**Comment by jongio** [comment]
@cxznmhdcxz - Please also add support for `... create --assign-identity`.  See spec for details: https://github.com/Azure/azure-cli/issues/9887#issuecomment-597195457

Thanks!
Jon

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 3]
```diff
@@ -281,6 +286,10 @@ def load_arguments(self, _):
         with self.argument_context(scope + ' identity') as c:
             c.argument('scope', help="The scope the managed identity has access to")
             c.argument('role', help="Role name or id the managed identity will be assigned")
+        with self.argument_context(scope + ' identity assign') as c:
+            c.argument('assign_identities', options_list=['--identities'], nargs='*', help="Space-separated identities to assign. Use '{0}' to refer to the system assigned identity. Default: '{0}'".format(MSI_LOCAL_ID))
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:290]:
```suggestion
            c.argument('assign_identities', options_list=['--identities'], nargs='*', help="Space-separated IDs of identities to assign. Use '{0}' to refer to the system assigned identity. Default: '{0}'".format(MSI_LOCAL_ID))
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 4]
```diff
@@ -281,6 +286,10 @@ def load_arguments(self, _):
...
+        with self.argument_context(scope + ' identity remove') as c:
+            c.argument('remove_identities', options_list=['--identities'], nargs='*', help="Space-separated identities to assign. Use '{0}' to refer to the system assigned identity. Default: '{0}'".format(MSI_LOCAL_ID))
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:292]:
```suggestion
            c.argument('remove_identities', options_list=['--identities'], nargs='*', help="Space-separated IDs of identities to assign. Use '{0}' to refer to the system assigned identity. Default: '{0}'".format(MSI_LOCAL_ID))
```

**Comment by qwordy** [comment]
How can I remove all identities now?

**Comment by qwordy** [comment]
Thanks for your description of this PR. Very clear!

**Comment by cxznmhdcxz** [comment]
> How can I remove all identities now?

@qwordy I'm afraid `identity remove` can't remove all identities now, unless one supply all currently assigned `--identities` 
 parameters.
`vm identity remove` can't remove all either.
Do we still need to support this?

**Comment by qwordy** [comment]
> 
> 
> > How can I remove all identities now?
> 
> @qwordy I'm afraid `identity remove` can't remove all identities now, unless one supply all currently assigned `--identities`
> parameters.
> `vm identity remove` can't remove all either.
> Do we still need to support this?

I am OK with the new behavior. But users may complain that the original function is broken. Can we remove all identities if --identities is not specified?

**Comment by jongio** [comment]
@cxznmhdcxz - Can you investigate why CI is failing?  Thanks, Jon

**Comment by jongio** [comment]
@tzhanl - Do we need help from the CLI team on the CI issue?

**Comment by panchagnula** [comment]
> > > How can I remove all identities now?
> > 
> > 
> > @qwordy I'm afraid `identity remove` can't remove all identities now, unless one supply all currently assigned `--identities`
> > parameters.
> > `vm identity remove` can't remove all either.
> > Do we still need to support this?
> 
> I am OK with the new behavior. But users may complain that the original function is broken. Can we remove all identities if --identities is not specified?

lets make sure we don't regress current behavior? @btardif do you have suggestions here if we should support removing all identities or not - or should we keep our behavior consistent with VM?

**Comment by btardif** [comment]
If ww are consistent with VM then we should be ok

> > > > How can I remove all identities now?
> > > 
> > > 
> > > @qwordy I'm afraid `identity remove` can't remove all identities now, unless one supply all currently assigned `--identities`
> > > parameters.
> > > `vm identity remove` can't remove all either.
> > > Do we still need to support this?
> > 
> > 
> > I am OK with the new behavior. But users may complain that the original function is broken. Can we remove all identities if --identities is not specified?
> 
> lets make sure we don't regress current behavior? @btardif do you have suggestions here if we should support removing all identities or not - or should we keep our behavior consistent with VM?

I'm ok with this since we are consistent with VM. We can revisit later if there is customer pushback.

Not having this today is blocking customer scenarios for ACR + App Service

**Comment by jongio** [comment]
Thanks @panchagnula and @qwordy and @cxznmhdcxz for implementing and reviewing this!  Great to see this get merged.

**Pull request merged**